keadm/cmd/keadm/app/cmd/config.go

```diff
@@ -0,0 +1,227 @@
+	cmd := &cobra.Command{
+		Use:   "config",
+		Short: "Use this command to configure keadm",
+		Long:  "Use this command to configure keadm",
+	}
+
+	cmd.AddCommand(newCmdConfigImages())
+	return cmd
+}
+
+// newCmdConfigImages returns the "keadm config images" command
+func newCmdConfigImages() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "images",
+		Short: "Interact with container images used by keadm",
+		Long:  "Use this command to `list/pull` keadm required container images",
+	}
+	cmd.AddCommand(newCmdConfigImagesList())
+	cmd.AddCommand(newCmdConfigImagesPull())
+	return cmd
+}
+
+// newCmdConfigImagesList returns the "keadm config images list" command
+func newCmdConfigImagesList() *cobra.Command {
+	cfg := newDefaultConfiguration()
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "Print a list of images keadm will use.",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			images := GetKubeEdgeImages(cfg)
+			for _, image := range images {
+				fmt.Println(image)
+			}
+
+			return nil
+		},
+		Args: cobra.NoArgs,
+	}
+
+	AddImagesCommonConfigFlags(cmd, cfg)
+	return cmd
+}
+
+// newCmdConfigImagesPull returns the `keadm config images pull` command
+// TODO: Now we only support docker images. If can, we will need to support more CRIs.
+func newCmdConfigImagesPull() *cobra.Command {
+	cfg := newDefaultConfiguration()
+
+	cmd := &cobra.Command{
+		Use:   "pull",
+		Short: "Pull images used by keadm",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			images := GetKubeEdgeImages(cfg)
+
+			return PullControlPlaneImages(images)
+		},
+		Args: cobra.NoArgs,
+	}
+	AddImagesCommonConfigFlags(cmd, cfg)
+
+	return cmd
+}
+
+// PullControlPlaneImages pulls all images that the ImagesPull knows about
+func PullControlPlaneImages(images []string) error {
+	for _, image := range images {
+		if err := dockerPullImage(image); err != nil {
+			return fmt.Errorf("failed to pull image %q: %v", image, err)
+		}
+		fmt.Printf("Successfully pulled %s\n", image)
+	}
+	return nil
+}
+
+// AddImagesCommonConfigFlags adds the flags that configure keadm
+func AddImagesCommonConfigFlags(cmd *cobra.Command, cfg *Configuration) {
+	cmd.Flags().StringVar(&cfg.KubeEdgeVersion, cmdcommon.KubeEdgeVersion, cfg.KubeEdgeVersion,
+		`Use this key to decide which a specific KubeEdge version to be used.`,
+	)
+	cmd.Flags().StringVar(&cfg.ImageRepository, cmdcommon.ImageRepository, cfg.ImageRepository,
+		`Use this key to decide which image repository to pull images from.`,
+	)
+	cmd.Flags().StringVar(&cfg.Component, "component", cfg.Component,
+		"Use this key to set whether is cloudcore or edgecore. If not set, keadm will list/pull all images used by both cloudcore and edgecore.")
+}
+
+// GetKubeEdgeImages returns a list of container images that related component expects to use
+func GetKubeEdgeImages(cfg *Configuration) []string {
+	var images []string
+
+	component := strings.ToLower(cfg.Component)
+
+	if component == "cloudcore" {
```

Comment:
Would it be better to divide it into `cloud` and `edge` sides here?

Reply:
yes, and what should cmd flag `--component` be fixed to? `--part`?

Reply:
`part`  looks good.

Reply:
how about --env?

Reply:
There can be some confusion with `--env`.
More like adding environment variables or setting the environment, such as dev or prod.
@zhu733756

Reply:
> There can be some confusion with `--env`. More like adding environment variables or setting the environment, such as dev or prod. @zhu733756

Yes, I fount it.

Reply:
> There can be some confusion with `--env`. More like adding environment variables or setting the environment, such as dev or prod. @zhu733756

Yes, I fount it.